PolyEngine/Engine/Src/World.cpp

```diff
@@ -2,14 +2,23 @@
 
 using namespace Poly;
 
-World::World(Poly::Engine* engine)
-: EntitiesAllocator(MAX_ENTITY_COUNT), EnginePtr(engine)
+World::World(Engine* engine, bool base)
+	: EntitiesAllocator(MAX_ENTITY_COUNT), EnginePtr(engine)
 {
 	memset(ComponentAllocators, 0, sizeof(IterablePoolAllocatorBase*) * MAX_COMPONENTS_COUNT);
+	memset(Components, 0, sizeof(ComponentBase*) * MAX_WORLD_COMPONENTS_COUNT);
+
+	if (!base)
```

Comment:
This whole block is unnecessary, beacause those components are added in Engine.hpp. Remove argument `base` from constructor.